wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/ast2wdlom/AstToTaskDefinitionElement.scala

```diff
@@ -1,12 +1,57 @@
 package wdl.draft3.transforms.ast2wdlom
 
-import common.Checked
-import common.validation.Checked._
+import cats.syntax.apply._
+import cats.syntax.validated._
+import cats.syntax.either._
+import common.collections.EnhancedCollections._
+import common.validation.ErrorOr._
 import wdl.draft3.parser.WdlParser.Ast
-import wdl.model.draft3.elements.TaskDefinitionElement
+import wdl.model.draft3.elements._
+import wdl.draft3.transforms.ast2wdlom.EnhancedDraft3Ast._
 
 object AstToTaskDefinitionElement {
 
-  def convert(a: Ast): Checked[TaskDefinitionElement] =
-    "FromAst[TaskDefinitionElement](a: Ast) is not implemented".invalidNelCheck
+  def convert(a: Ast): ErrorOr[TaskDefinitionElement] = {
+    val nameElementValidation: ErrorOr[String] = astNodeToString(a.getAttribute("name")).toValidated
+    val sectionsValidation: ErrorOr[Vector[TaskSectionElement]] = a.getAttributeAsVector[TaskSectionElement]("declarations").toValidated
+
+    (nameElementValidation, sectionsValidation) flatMapN combineElements
+  }
+
+  def combineElements(nameElement: String, bodyElements: Vector[TaskSectionElement]) = {
+    val inputsSectionElement: ErrorOr[Option[InputsSectionElement]] = validatedSizeEquals(bodyElements.filterByType[InputsSectionElement], "inputs", 1)
+    val outputsSectionElement: ErrorOr[Option[OutputsSectionElement]] = validatedSizeEquals(bodyElements.filterByType[OutputsSectionElement], "outputs", 1)
+    val runtimeSectionElement: ErrorOr[Option[RuntimeAttributesSectionElement]] = validatedSizeEquals(bodyElements.filterByType[RuntimeAttributesSectionElement], "runtime", 1)
```

Comment:
Sorry, I think my last comment was misleading... these elements are `validateSizeIsLessOrEqual `, but the `command` section will be `validatedSizeEquals `